#
# old_revision [f72a9682370efd023ef22aa4447b4dc182840216]
#
# patch "mtn-browse"
#  from [54b0d5d06be650637bb5b14a9d357ea8451a141c]
#    to [911314dd989170042eec6970a4a1139575e45ec4]
#
============================================================
--- mtn-browse	54b0d5d06be650637bb5b14a9d357ea8451a141c
+++ mtn-browse	911314dd989170042eec6970a4a1139575e45ec4
@@ -77,7 +77,7 @@ use Gtk2::Pango;
 use Gtk2::GladeXML;
 use Gtk2::Helper;
 use Gtk2::Pango;
-use Gtk2::SourceView;
+use Gtk2::SourceView2;
 use IO::Dir;
 use IO::File;
 use IPC::Open3;
@@ -2247,13 +2247,13 @@ sub get_browser_window(;$$$$$)
 
         # Setup the file file viewer (with syntax highlighting).
 
-        $browser->{file_view_svbuffer} = Gtk2::SourceView::Buffer->new(undef);
-        $browser->{file_view_svbuffer}->set_check_brackets(FALSE);
+        $browser->{file_view_svbuffer} = Gtk2::SourceView2::Buffer->new(undef);
+        $browser->{file_view_svbuffer}->set_highlight_matching_brackets(FALSE);
         $browser->{file_view_svbuffer}->set_max_undo_levels(0);
         $browser->{file_view_svbuffer}->begin_not_undoable_action();
         $browser->{file_view_svlangmgr} =
-            Gtk2::SourceView::LanguagesManager->new();
-        $browser->{file_view_sv} = Gtk2::SourceView::View->
+            Gtk2::SourceView2::LanguageManager->new();
+        $browser->{file_view_sv} = Gtk2::SourceView2::View->
             new_with_buffer($browser->{file_view_svbuffer});
         $browser->{file_view_sv}->modify_font($mono_font);
         $browser->{file_view_sv}->set_cursor_visible(FALSE);
@@ -3344,7 +3344,7 @@ sub update_browser_state($$)
             $browser->{file_view_svbuffer}->
                 place_cursor($browser->{file_view_svbuffer}->get_start_iter());
             $browser->{file_view_svbuffer}->set_text("");
-            $browser->{file_view_svbuffer}->set("highlight", FALSE);
+            $browser->{file_view_svbuffer}->set_highlight_syntax(FALSE);
             set_label_value($browser->{file_name_value_label}, "");
             set_label_value($browser->{file_author_value_label}, "");
             set_label_value($browser->{last_update_value_label}, "");
@@ -3469,7 +3469,7 @@ sub display_file($$)
     $browser->{file_view_svbuffer}->
         place_cursor($browser->{file_view_svbuffer}->get_start_iter());
     $browser->{file_view_svbuffer}->set_text("");
-    $browser->{file_view_svbuffer}->set("highlight", FALSE);
+    $browser->{file_view_svbuffer}->set_highlight_syntax(FALSE);
 
     # Get contents.
 
@@ -3573,9 +3573,12 @@ sub display_file($$)
                          && $mime_details->{syntax_highlight}))
                     && defined($lang =
                                $browser->{file_view_svlangmgr}->
-                                   get_language_from_mime_type($mime_type)))
+                                   guess_language($browser->
+                                                      {file_being_viewed}->
+                                                      {short_name},
+                                                  $mime_type)))
                 {
-                    $browser->{file_view_svbuffer}->set("highlight", TRUE);
+                    $browser->{file_view_svbuffer}->set_highlight_syntax(TRUE);
                     $browser->{file_view_svbuffer}->set_language($lang);
                 }
 
